var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -82,49 +94,89 @@ class Gcc(AutotoolsPackage):
+    # run the following command in the GCC source directory:
+    #    $ grep ^language= gcc/*/config-lang.in
+    # See https://gcc.gnu.org/install/configure.html
+
+    # Support for processing BRIG 1.0 files was added in GCC 7
+    # BRIG is a binary format for HSAIL:
+    # (Heterogeneous System Architecture Intermediate Language).
+    # See https://gcc.gnu.org/gcc-7/changes.html
+    conflicts('languages=brig', when='@:6')
+
+    # GCC 4.8 added a 'c' language. I'm sure C was always built,
+    # but this is the first version that accepts 'c' as a valid language.
+    conflicts('languages=c', when='@:4.7')
+
+    # GCC 4.6 added support for the Go programming language.
+    # See https://gcc.gnu.org/gcc-4.6/changes.html
+    conflicts('languages=go', when='@:4.5')
+
+    # The GCC Java frontend and associated libjava runtime library
+    # have been removed from GCC as of GCC 7.
+    # See https://gcc.gnu.org/gcc-7/changes.html
+    conflicts('languages=java', when='@7:')
+
+    # GCC 5 added the ability to build GCC as a Just-In-Time compiler.
+    # See https://gcc.gnu.org/gcc-5/changes.html
+    conflicts('languages=jit', when='@:4')
+
+    patch('darwin/gcc-7.1.0-headerpad.patch', when='@7.1.0 platform=darwin')
+    patch('darwin/gcc-6.1.0-jit.patch', when='@6.1.0 platform=darwin')
+    patch('darwin/gcc-4.9.patch1', when='@4.9.0:4.9.3 platform=darwin')
+    patch('darwin/gcc-4.9.patch2', when='@4.9.0:4.9.3 platform=darwin')
     patch('piclibs.patch', when='+piclibs')
     patch('gcc-backport.patch', when='@4.7:4.9.2,5:5.3')
 
-    def configure_args(self):
+    def patch(self):
         spec = self.spec
         prefix = self.spec.prefix
 
-        enabled_languages = set(('c', 'c++', 'fortran', 'java', 'objc'))
-
-        if spec.satisfies("@4.7.1:") and sys.platform != 'darwin' and \
-           not (spec.satisfies('@:4.9.3') and 'ppc64le' in spec.architecture):
-            enabled_languages.add('go')
-
         # Fix a standard header file for OS X Yosemite that
         # is GCC incompatible by replacing non-GCC compliant macros
         if 'yosemite' in spec.architecture:
-            if isfile(r'/usr/include/dispatch/object.h'):
+            if isfile('/usr/include/dispatch/object.h'):
                 new_dispatch_dir = join_path(prefix, 'include', 'dispatch')
                 mkdirp(new_dispatch_dir)
                 cp = which('cp')
                 new_header = join_path(new_dispatch_dir, 'object.h')
-                cp(r'/usr/include/dispatch/object.h', new_header)
+                cp('/usr/include/dispatch/object.h', new_header)
                 filter_file(r'typedef void \(\^dispatch_block_t\)\(void\)',
                             'typedef void* dispatch_block_t',
                             new_header)
 
+    def configure_args(self):
+        spec = self.spec
+
         # Generic options to compile GCC
         options = [
-            '--libdir={0}'.format(prefix.lib64),
```

Comment:
Probably because of multilib and 32bit systems, we had similar problems in hashdist as well with where to pub the libs, it can cause linking issues when the compiler is built.